tasks/sql-connector/main.py

```diff
@@ -0,0 +1,118 @@
+from typing import Any, Union
+
+import logging
+import csv
+import zipfile
+
+from mysql.connector import CMySQLConnection
+from psycopg2.extensions import connection
+
+import mysql.connector
+import psycopg2
+
+from coretex import currentTaskRun, CustomDataset
+
+
+def connectMysqlDatabase(connectionConfig: dict[str, str]) -> CMySQLConnection:
+    logging.info(f">> [SQL Connector] Connecting with MySQL database \"{connectionConfig["database"]}\"...")
+
+    try:
+        conn = mysql.connector.connect(**connectionConfig)
+    except mysql.connector.errors.Error as e:
+        logging.error(f">> [SQL Connector] Error while connecting to database: {e}")
+
+    return conn
+
+
+def connectPostgresqlDatabase(connectionConfig: dict[str, str]) -> connection:
+    logging.info(f">> [SQL Connector] Connecting with PostgreSQL database \"{connectionConfig["database"]}\"...")
+
+    try:
+        conn = psycopg2.connect(**connectionConfig)
+    except psycopg2._psycopg.Error as e:
+        logging.error(f">> [SQL Connector] Error while connecting to database: {e}")
+
+    return conn
+
+
+def fetchAllData(conn: Any, dataset: CustomDataset, queryGetTables: str, queryGetRows: str) -> None:
+    cursor = conn.cursor()
+    cursor.execute(queryGetTables)
+    tables = cursor.fetchall()
+    tables = [table[0] for table in tables]
+
+    for table in tables:
+        tableData: list[dict[str, str]] = []
+
+        cursor.execute(queryGetRows + f"'{table}'")
+        columnNames = list(cursor.fetchall())
+        columnNames = [name[0] for name in columnNames]
+
+        cursor.execute(f"SELECT * FROM {table}")
+        rows = list(cursor.fetchall())
+
+        for row in rows:
+            tableData.append(dict(zip(columnNames, list(row))))
+
+        sampleNameCSV = f"{table}.csv"
+        with open(sampleNameCSV, "w", newline="") as file:
+            writer = csv.DictWriter(file, fieldnames = columnNames)
+            writer.writeheader()
+            writer.writerows(tableData)
+
+        sampleNameZIP = f"{table}.zip"
```

Comment:
We use camelCase style so `sampleNameCSV` -> `sampleNameCsv`, `sampleNameZIP` -> `sampleNameZip`